AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -300,13 +305,19 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
         syncEventEmitter = SyncEventEmitter(initialSyncOrchestrator: initialSyncOrchestrator,
                                             reconciliationQueue: reconciliationQueue)
 
-        readyEventEmitter = ReadyEventEmitter(remoteSyncEnginePublisher: publisher,
-                                              completion: { self.cancelEmitters() })
+        readyEventEmitter = ReadyEventEmitter(
+            remoteSyncEnginePublisher: publisher,
+            completion: { [weak self] in self?.cancelEmitters() }
+        )
 
         // TODO: This should be an AsynchronousOperation, not a semaphore-waited block
         let semaphore = DispatchSemaphore(value: 0)
 
-        initialSyncOrchestrator.sync { result in
+        initialSyncOrchestrator.sync { [weak self] result in
+            guard let self = self else {
+                return
```

Comment:
is it safe to return early here considering that we signal the semaphore on line #335?

Reply:
Yikes, good callout. I don't know what happens if we lose `self` while we have a waited semaphore, but I'll fix.